File: java/darts/src/main/java/Darts.java
class Darts {
    private double distance;

    Darts(double x, double y) {
        this.distance = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
    }

    int score() {
        if (distance <= 1.0) {
            return 10;
        } else if (distance <= 5.0) {
            return 5;
        } else if (distance <= 10.0) {
            return 1;
        } else {
            return 0;
        }
    }
}

File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.sqrt

object Darts {
    fun score(x: Int, y: Int) = score(x.toDouble(), y.toDouble())
    fun score(x: Int, y: Double) = score(x.toDouble(), y)
    fun score(x: Double, y: Int) = score(x, y.toDouble())

    fun score(x: Double, y: Double): Int {
        val dist = distance(x, y)
        return when {
            dist > 10.0 -> 0
            dist > 5.0 -> 1
            dist > 1.0 -> 5
            else -> 10
        }
    }

    private fun distance(x: Double, y: Double): Double = sqrt(x * x + y * y)
}
